### PR TITLE
Fix for AKS recent provider change

### PR DESCRIPTION
**Pull request by aLekSer**
Remove deprecation warnings for AKS and EKS syntax : "${}"

About 6 days ago there was an update of AzureRM terraform provider, which broke AKS terraform config:
https://stackoverflow.com/questions/60384689/terraform-azurerm-2-x-error-features-required-field-is-not-set

### CODE REVIEW

**Comment by aLekSer** [comment]
This is the ticket 
https://github.com/terraform-providers/terraform-provider-azurerm/issues/5867

**Comment by aLekSer** [comment]
This PR is ready for review. 
There are 2 commented out lines.

I will add this in an upcoming release :
```
enable_node_public_ip = true
```
(current approach with opening a PublicIP works here as well https://agones.dev/site/docs/installation/creating-cluster/aks/#creating-and-assigning-public-ips-to-nodes )

I have noticed strange error message in build e679ada4-..
e679ada4-
```
Step #22: time="2020-03-03 14:17:49.151" level=info msg="fleet simple-fleet-tnwnh has 3/3 ready replicas"
Step #22: time="2020-03-03 14:17:49.213" level=info msg="failing PostAllocate request" error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: authentication handshake failed: x509: cannot validate certificate for 35.247.4.172 because it doesn't contain any IP SANs\""
Step #22: time="2020-03-03 14:17:56.022" level=info msg="waiting for fleet condition" fleet=simple-fleet-v8jd9
Step #22: time="2020-03-03 14:17:56.054" level=info msg="fleet simple-fleet-v8jd9 has 0/3 ready replicas"
Step #22: time="2020-03-03 14:17:58.062" level=info msg="fleet simple-fleet-v8jd9 has 0/3 ready replicas"
```

File: install/terraform/modules/aks/aks.tf [hunk 1]
```diff
@@ -45,33 +53,58 @@ resource "azurerm_resource_group" "test" {
...
+  name                  = "system"
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.test.id
+  vm_size               = var.machine_type
+  node_count            = 1
+  os_disk_size_gb       = var.disk_size
+  enable_auto_scaling   = false
+  node_taints           = ["agones.dev/agones-system=true:NoExecute"]
+}
+
+/*
```
Comment (markmandel) [inline install/terraform/modules/aks/aks.tf]: Nit: remove commented out code.

Comment (aLekSer) [reply install/terraform/modules/aks/aks.tf]: Sorry, this code was added due to my quota (of 4 VMs) limit has reached.
Tested with:
```
 terraform apply -var client_id=cb... -var client_secret=9e... -var node_count=1 --auto-approve
```
All is fine with this Metrics block also.

**Comment by aLekSer** [comment]
Bingo situation: 3 times `TestGameServerReserve` failures in a row, hope that it would be fixed soon.

File: install/terraform/modules/aks/aks.tf [hunk 2]
```diff
@@ -45,33 +53,56 @@ resource "azurerm_resource_group" "test" {
 }
 
 resource "azurerm_kubernetes_cluster" "test" {
```
[the remark this answers is not in the document]
Comment (aLekSer) [reply install/terraform/modules/aks/aks.tf]: This caused by copying of example code of azure terraform provider

**Pull request merged**